src/lib.rs

```diff
@@ -1674,6 +1674,64 @@ impl<B: BitBlock> BitVec<B> {
             carry = curr_carry;
         }
     }
+
+    /// Appends an element if there is sufficient spare capacity, otherwise an error is returned
+    /// with the element.
+    ///
+    /// Unlike [`push`] this method will not reallocate when there's insufficient capacity.
+    /// The caller should use [`reserve`] to ensure that there is enough capacity.
+    ///
+    /// [`push`]: Self::push
+    /// [`reserve`]: Self::reserve
+    ///
+    /// # Examples
+    /// ```
+    /// use bit_vec::BitVec;
+    ///
+    /// let mut bitvec = BitVec::from_elem(63, false); // there is space for exactly one element
+    ///
+    /// assert!(bitvec.push_within_capacity(true).is_ok()); // successfully push
+    /// assert_eq!(bitvec.get(63), Some(true));
+    ///
+    /// assert_eq!(bitvec.len(), bitvec.capacity()); // no spare capacity
```

Comment:
If we, say, change the way we initialize bit-vecs, this will begin to fail. Ideally please do not include this implementation detail irrelevant for the feature in a code that's read and executed for testing.

I would just save capacity after the first line and compare it here
```rust
let initial_capacity = bitvec.capacity();
...
assert_eq!(bitvec.capacity(), initial_capacity);
```

Reply:
Oh! Best use `BitVec::with_capacity(64)` to show which capacity we're really working with